### DISCUSSION TITLE
copilot cleanup fails - where are environment details kept ?

### DISCUSSION THREAD

**Discussion by paulhhallam** [aws/copilot-cli#2488] [Q&A] [answered]
Attempting to use Jenkins for CICD I kept going round and round in circles and finally decided to clean up my environment.
(the EC2 host was using a differnet account to the Jenkins AWS connection)
I kept getting permission errors so decided to clean up and try to work through methodically.
I used the copilot svc delete  etc to delete the service environments but had the same permission errors.
I then went into AWS and used the cloudformation delete stacks to get everything clean. 
The VPC etc has been removed. One issue was it complained it couldn't finish because the S3 bucket had files in it so I cleaned that up and ran the delete stack again.
BUT 
back on my EC2 linux host even though I have removed the copilot directory 
"copilot app ls" still lists 2 apps.
"copilot svc ls" still lists 1 service.
"copilot env ls" still lists 1 environment.
Can anyone tell me what else I have to do to get a clean environment.

Many thanks in advance

**Comment by paulhhallam** [comment, marked as answer]
OK, I'm an idiot - its all in SSM parameter store.
Guess I need to sit down and read the whole manual.

**Reply by efekarakus** [reply]
Hi @paulhhallam ! I think we can do a better job there pointing at maybe where we fetch that metadata.

Glad you managed to find out that SSM parameter store is where we store this info! Hope that unblocked you 🙇
